docs/guides/cross-referencing-with-sphinx.rst

```diff
@@ -0,0 +1,278 @@
+Cross-referencing with Sphinx
+=============================
+
+When writing documentation you often need to link to other pages of your documentation,
+or other sections of the current page, or sections from other pages.
+
+.. _target to paragraph:
+
+An easy way is just to use the final link of the page/section.
+This works, but it has some disadvantages:
+
+- Links can change, so they are hard to maintain.
+- Links can be verbose, not easy to know the page/section they are linking to.
+- Not easy way to link to specific sections like paragraphs, figures, or code blocks.
+- It only works for the html version of your documentation.
+
+ReStructuredText has a built-in way to linking to elements,
```

Comment:
Shouldn't this be `docutils` as opposed to `reStructuredText`? I may or may not be sensitive to this because MyST markdown also uses the same docutils cross-referencing :-)